packages/cashier/src/modules/deposit-fiat/components/deposit-fiat-iframe/deposit-fiat-iframe.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import { Loading } from '@deriv/components';
+import { useDepositFiatAddress } from '@deriv/hooks';
+import { observer } from '@deriv/stores';
+import './deposit-fiat-iframe.scss';
+
+const DepositFiatIframe: React.FC = observer(() => {
+    const { data: iframe_url, isSuccess } = useDepositFiatAddress();
+    const [is_loading, setIsLoading] = useState(true);
+
+    React.useEffect(() => {
+        setIsLoading(true);
+    }, [iframe_url]);
+
+    return (
+        <React.Fragment>
+            {is_loading && <Loading className='deposit-fiat-iframe__loader' is_fullscreen={false} />}
+            {isSuccess && (
+                <iframe
```

Comment:
```suggestion
                <iframe
                    key={iframe_url}
```
@aum-deriv We need to use the URL as a key here to make sure when the theme changes and we have a new URL we don't show the loaded page in the iframe while we are loading the new URL.